workspace-loader/src/index.ts

```diff
@@ -225,33 +208,34 @@ export class WorkspaceLoader {
                 xhr.onreadystatechange = () => {
                     if (xhr.readyState !== 4) { return; }
                     if (xhr.status !== 200) {
-                        let response;
-                        try {
-                            response = JSON.parse(xhr.responseText);
-                        } catch (e) {}
-
-                        if (response) {
-                            reject(response);
-                        } else if (xhr.statusText) {
-                            reject(xhr.statusText);
-                        } else {
-                            reject("Unknown error");
-                        }
+                        const errorMessage = 'Failed to start the workspace'  + this.getRequestErrorMessage(xhr);
+                        reject(errorMessage);
                         return;
                     }
                     resolve(JSON.parse(xhr.responseText));
                 };
             });
-        }).catch((error: any) => {
-            let errorMessage = '';
-            if (error) {
-                errorMessage = error.message ? error.message.toString() : error.toString();
-            }
-            errorMessage = `Failed to start the workspace` + (errorMessage ? `: "${errorMessage}"` : '.');
-            return Promise.reject(errorMessage);
         });
     }
 
+    getRequestErrorMessage(xhr: XMLHttpRequest): string {
+        let errorMessage;
+        try {
+            const response = JSON.parse(xhr.responseText);
+            errorMessage = response.message;
+        } catch (e) { }
```

Comment:
should we log error message/response ?

Reply:
It is eventually logged here https://github.com/eclipse/che/blob/389e259d7672fe57300f10ad19657076ea7f19f9/workspace-loader/src/index.ts#L145-L146

Here I only tried to get a meaningful error message to reject a workspace getting or starting promise.